test/buffer-test.js

```diff
@@ -82,5 +83,88 @@ module.exports = function (repo) {
         })
       })
     })
+
+    it('export a file with no links', (done) => {
+      const hash = 'QmQmZQxSKQppbsWfVzBvg59Cn3DKtsNVQ94bjAxg2h3Lb8'
+      const bs = new BlockService(repo)
+      const ds = new DAGService(bs)
+      const testExport = importer.export(hash, ds)
+      testExport.on('file', (data) => {
+        ds.get(hash, (err, fetchedNode) => {
+          expect(err).to.not.exist
+          const unmarsh = UnixFS.unmarshal(fetchedNode.data)
+          expect(unmarsh.data).to.deep.equal(data.stream._readableState.buffer[0])
+          done()
+        })
+      })
+    })
+
+    it('export a small file with links', (done) => {
+      const hash = 'QmW7BDxEbGqxxSYVtn3peNPQgdDXbWkoQ6J1EFYAEuQV3Q'
+      const bs = new BlockService(repo)
+      const ds = new DAGService(bs)
+      const testExport = importer.export(hash, ds)
+      testExport.on('file', (data) => {
+        ds.get(hash, (err, fetchedNode) => {
+          expect(err).to.not.exist
+          var i = 0
+          async.forEachSeries(fetchedNode.links, (link, callback) => {
+            ds.get(link.hash, (err, res) => {
+              expect(err).to.not.exist
+              const unmarsh = UnixFS.unmarshal(res.data)
+              expect(unmarsh.data).to.deep.equal(data.stream._readableState.buffer[i])
+              i++
+              callback()
+            })
+          }, (err) => {
+            expect(err).to.not.exist
+            done()
+          })
+        })
+      })
+    })
+
+    it('export a large file > 5mb', (done) => {
+      const hash = 'QmRQgufjp9vLE8XK2LGKZSsPCFCF6e4iynCQtNB5X2HBKE'
+      const bs = new BlockService(repo)
+      const ds = new DAGService(bs)
+      const testExport = importer.export(hash, ds)
+      testExport.on('file', (data) => {
+        ds.get(hash, (err, fetchedNode) => {
+          expect(err).to.not.exist
+          var i = 0
+          async.forEachSeries(fetchedNode.links, (link, callback) => {
+            ds.get(link.hash, (err, res) => {
+              expect(err).to.not.exist
+              const unmarsh = UnixFS.unmarshal(res.data)
+              expect(unmarsh.data).to.deep.equal(data.stream._readableState.buffer[i])
+              i++
```

Comment:
the i is incrementing the index on the array of buffer that build up in the stream object that gets passed back to make sure that they are in the correct order. Matching the order in the dag links as the async series steps the through the dag links